baiji/serialization/json.py

```diff
@@ -12,15 +12,73 @@ def load(f, *args, **kwargs):
     return ensure_file_open_and_call(f, _load, 'r', *args, **kwargs)
 
 def loads(*args, **kwargs):
+    kwargs.update(object_hook=_json_decode)
     return json.loads(*args, **kwargs)
 
 def dumps(*args, **kwargs):
-    kwargs.update(for_json=True)
+    kwargs.update(for_json=True, default=_json_encode)
     return json.dumps(*args, **kwargs)
 
 def _dump(f, obj, *args, **kwargs):
-    kwargs.update(for_json=True)
+    kwargs.update(for_json=True, default=_json_encode)
     return json.dump(obj, f, *args, **kwargs)
 
 def _load(f, *args, **kwargs):
+    kwargs.update(object_hook=_json_decode)
     return json.load(f, *args, **kwargs)
+
+def _json_decode(dct):
+    '''
+    Handle Custom json decoding and internal double underscore annotated Direct object deserialization
+    '''
+    if '__ndarray__' in dct:
+        if not 'dtype' in dct and 'shape' in dct:
```

Comment:
Tricks always seems to write those two and it isn't clear that it can cope with them being missing. But I can get rid of them easily enough, just in case someone writes something weird.